test/test_stat.rs

```diff
@@ -260,3 +261,36 @@ fn test_utimensat() {
               UtimensatFlags::FollowSymlink).unwrap();
     assert_times_eq(500, 800, &fs::metadata(&fullpath).unwrap());
 }
+
+#[test]
+fn test_mkdirat_success_path() {
+    let tempdir = tempfile::tempdir().unwrap();
+    let filename = "example_subdir";
+    let dirfd = fcntl::open(tempdir.path(), fcntl::OFlag::empty(), stat::Mode::empty()).unwrap();
+    assert!((mkdirat(dirfd, filename, Mode::S_IRWXU)).is_ok());
+    assert!(Path::exists(&tempdir.path().join(filename)));
+}
+
+#[test]
+fn test_mkdirat_success_mode() {
+    let expected_bits = stat::SFlag::S_IFDIR.bits() | stat::Mode::S_IRWXU.bits();
+    let tempdir = tempfile::tempdir().unwrap();
+    let filename = "example_subdir";
+    let dirfd = fcntl::open(tempdir.path(), fcntl::OFlag::empty(), stat::Mode::empty()).unwrap();
+    assert!((mkdirat(dirfd, filename, Mode::S_IRWXU)).is_ok());
+    let permissions = fs::metadata(tempdir.path().join(filename)).unwrap().permissions();
+    let mode = permissions.mode();
+    assert_eq!(mode as mode_t, expected_bits)
+}
+
+#[test]
+#[should_panic="EACCES"]
+fn test_mkdirat_fail() {
+    let tempdir = tempfile::tempdir().unwrap();
+    let filename = "example_subdir";
+    let mut perms = fs::metadata(tempdir.path()).unwrap().permissions();
+    perms.set_readonly(true);
+    fs::set_permissions(tempdir.path(), perms).unwrap();
+    let dirfd = fcntl::open(tempdir.path(), fcntl::OFlag::empty(), stat::Mode::empty()).unwrap();
+    let result = mkdirat(dirfd, filename, Mode::S_IRWXU).unwrap();
```

Comment:
Instead of using `should_panic`, you can just replace `unwrap` with `unwrap_err`.